src/lib/RuleFilter.ts

```diff
@@ -135,7 +131,19 @@ abstract class NegateableRuleFilter extends RuleFilter {
 		}
 
 		const negated = mapped.negative.length > 0;
+
 		super(negated ? mapped.negative : mapped.positive, negated);
+		this.usingDefaultEngines = usingDefaultEngines;
+	}
+
+	public matchesRule(rule: Rule): boolean {
+		// If this rule is negated, we need to make sure to not match if it not on by default while we are using
+		// the default engine. If using specified (non-default) engine list, then the defaultEnabled flag is irrelevant.
+		if (this.negated && this.usingDefaultEngines && !rule.defaultEnabled) {
```

Comment:
I think technically we could return false for all RuleFilters (not just Negatable ones) whenever this.usingDefaultEngines && !rule.defaultEnabled. But it only impacts the category filter... so just limiting this check to this situation for now.